receiver/otlpreceiver/otlp.go

```diff
@@ -80,7 +80,9 @@ func New(
 	r := &Receiver{
 		ln:          ln,
 		corsOrigins: []string{}, // Disable CORS by default.
-		gatewayMux:  gatewayruntime.NewServeMux(),
+		gatewayMux: gatewayruntime.NewServeMux(
+			gatewayruntime.WithMarshalerOption("application/x-protobuf", &xProtobufMarshaler{}),
```

Comment:
Should this be controlled by a config?

Reply:
I think it should be fine being on by default since it's just adding support for a new content-type to the existing HTTP server. When gzipped responses are added, though, maybe it could be useful to have the option to enable or disable them.

Reply:
I think it is fine to enable all transports in the receiver. I do not see why we would want to reject OTLP/HTTP incoming requests. However if we see that need for it we can add a config option in the future.

Reply:
I feel that defaults are fine but no option to disable, which may not be a problem now.